src/Symfony/Component/Yaml/Command/LintCommand.php

```diff
@@ -137,40 +143,75 @@ private function display(SymfonyStyle $io, array $files): int
                 throw new InvalidArgumentException(sprintf('The format "%s" is not supported.', $this->format));
         }
     }
 
-    private function displayTxt(SymfonyStyle $io, array $filesInfo): int
+    private function displayTxt(SymfonyStyle $io, array $filesInfo, bool $errorAsGithubAnnotations = false): int
     {
         $countFiles = \count($filesInfo);
         $erroredFiles = 0;
         $suggestTagOption = false;
 
+        $buffer = [];
         foreach ($filesInfo as $info) {
             if ($info['valid'] && $this->displayCorrectFiles) {
                 $io->comment('<info>OK</info>'.($info['file'] ? sprintf(' in %s', $info['file']) : ''));
             } elseif (!$info['valid']) {
                 ++$erroredFiles;
-                $io->text('<error> ERROR </error>'.($info['file'] ? sprintf(' in %s', $info['file']) : ''));
-                $io->text(sprintf('<error> >> %s</error>', $info['message']));
 
                 if (false !== strpos($info['message'], 'PARSE_CUSTOM_TAGS')) {
                     $suggestTagOption = true;
                 }
+
+                if ($errorAsGithubAnnotations) {
+                    $buffer[] = $this->formatErrorAsGithubAnnotation($info);
+                    continue;
+                }
+
+                $io->text('<error> ERROR </error>'.($info['file'] ? sprintf(' in %s', $info['file']) : ''));
+                $io->text(sprintf('<error> >> %s</error>', $info['message']));
             }
         }
 
         if (0 === $erroredFiles) {
             $io->success(sprintf('All %d YAML files contain valid syntax.', $countFiles));
         } else {
-            $io->warning(sprintf('%d YAML files have valid syntax and %d contain errors.%s', $countFiles - $erroredFiles, $erroredFiles, $suggestTagOption ? ' Use the --parse-tags option if you want parse custom tags.' : ''));
+            $message = sprintf('%d YAML files have valid syntax and %d contain errors.%s', $countFiles - $erroredFiles, $erroredFiles, $suggestTagOption ? ' Use the --parse-tags option if you want parse custom tags.' : '');
+
+            if ($errorAsGithubAnnotations) {
+                $io->writeln(sprintf('::warning::%s', $message));
+                $io->writeln($buffer);
+            } else {
+                $io->warning($message);
+            }
         }
 
         return min($erroredFiles, 1);
     }
 
+    private function formatErrorAsGithubAnnotation(array $info): string
+    {
+        // newlines need to be encoded
+        // see https://github.com/actions/starter-workflows/issues/68#issuecomment-581479448
+        $message = str_replace("\n", '%0A', $info['message']);
+
+        if (!$info['file']) {
+            return sprintf('::error::%s', $message);
+        }
+
+        $metas = ['file' => $info['file'], 'line' => $info['line'], 'col' => 0];
+
+        array_walk($metas, static function (&$value, string $key): void {
+            $value = sprintf('%s=%s', $key, (string) $value);
```

Comment:
this needs to escape values with the `escapeProperty` logic of the official toolkit